src/main/java/hudson/tasks/test/TestResult.java

```diff
@@ -276,19 +278,25 @@ public String toPrettyString() {
     }
 
     /**
-     * Annotate some text -- what does this do? 
+     * Annotate text by turning it into a clickable link and then sanitizing it using owasp-java-html-sanitizer
+     * @see <a href="https://www.javadoc.io/doc/com.googlecode.owasp-java-html-sanitizer/owasp-java-html-sanitizer/20160628.1/org/owasp/html/HtmlPolicyBuilder.html">owasp HtmlPolicyBuilder javadoc</a>
      * @param text Text to use to annotate the actions.
      *
      * @return the provided text HTML-escaped.
      */
     public String annotate(String text) {
         if (text == null)
                 return null;
-        text = text.replace("&", "&amp;").replace("<", "&lt;");
 
-        for (TestAction action: getTestActions()) {
-                text = action.annotate(text);
-        }
-        return text;
+        text = text.replace("&", "&amp;").replace("<", "&lt;").replaceAll("\\b(https?://[^\\s)>]+)", "<a href=\"$1\">$1</a>");
```

Comment:
Could you add a test to confirm that `https://javascript:alert(1)` getting transformed into `<a href="javascript:alert(1)"` will correctly be caught by the policy?